SMBLibrary/Client/ConnectionState.cs

```diff
@@ -4,41 +4,34 @@
  * the GNU Lesser Public License as published by the Free Software Foundation,
  * either version 3 of the License, or (at your option) any later version.
  */
+
 using System;
-using System.Collections.Generic;
-using System.Net;
 using System.Net.Sockets;
 using SMBLibrary.NetBios;
-using SMBLibrary.SMB1;
-using Utilities;
 
 namespace SMBLibrary.Client
 {
-    public class ConnectionState
+    public sealed class ConnectionState : IDisposable
     {
-        private Socket m_clientSocket;
-        private NBTConnectionReceiveBuffer m_receiveBuffer;
+        private readonly Socket m_clientSocket;
+        private readonly NBTConnectionReceiveBuffer m_receiveBuffer;
 
         public ConnectionState(Socket clientSocket)
         {
             m_clientSocket = clientSocket;
             m_receiveBuffer = new NBTConnectionReceiveBuffer();
         }
 
-        public Socket ClientSocket
-        {
-            get
-            {
-                return m_clientSocket;
-            }
-        }
+        public Socket ClientSocket => m_clientSocket;

```

Comment:
How are those changes related to buffer reuse?
It's best to refrain from making unrelated changes in a single PR that shift focus from what you're trying to achieve